docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -845,3 +845,22 @@ function _su_humsci_profile_get_uuid(string $config_name): string|null {
   $config = $config_storage->read($config_name);
   return $config['uuid'] ?? NULL;
 }
+
+/**
+* Uninstall legacy themes.
+*/
+function su_humsci_profile_update_9705() {
+  $extension_config = \Drupal::service('config.factory')->get('core.extension');
+  $legacy_themes = [
+    'archaeology',
+    'francestanford',
+    'mathematics',
+    'su_humsci_subtheme',
+    'su_humsci_theme',
+  ];
+  // Filter out the themes that are not installed.
+  $installed_themes = array_filter($legacy_themes, function ($theme) use ($extension_config) {
+    return $extension_config->get("theme.{$theme}") !== NULL;
+  });
```

Comment:
Why not just use `\Drupal::service('extension.list.theme)->getList()` to get a list of installed themes?

Reply:
@joegl Thanks for the feedback! I just implemented the change, just replaced the `getList()` method for `getAllInstalledInfo()`, because `getList()` returns all available themes, not only the installed ones.